### PR TITLE
NLog SentryTarget with less overhead for breadcrumb

### PR DESCRIPTION
**Pull request by snakefoot**
When IncludeEventDataOnBreadcrumbs = false (Default)

### CODE REVIEW

**Comment by bruno-garcia** [comment]
Thanks for your contribution @snakefoot!
You change introduced a couple of branches uncovered by tests. Could you please test those out before we can merge it?

File: src/Sentry.Log4Net/Sentry.Log4Net.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>netstandard2.0;net461</TargetFrameworks>
```
Comment (bruno-garcia) [inline src/Sentry.Log4Net/Sentry.Log4Net.csproj]: Unrelated to NLog change. Ideally would go on a separate PR but I'm happy to merge it in.

File: src/Sentry.NLog/LogLevelExtensions.cs [hunk 2]
```diff
@@ -29,7 +29,7 @@ internal static class LogLevelExtensions
                     return SentryLevel.Warning;
 
                 default:
-                    return null;
+                    return SentryLevel.Info;
```
Comment (bruno-garcia) [inline src/Sentry.NLog/LogLevelExtensions.cs]: Don't see any tests dealing with the change of behavior here. Not sure what is the impact of this change now.

File: src/Sentry.NLog/SentryTarget.cs [hunk 3]
```diff
@@ -259,12 +259,11 @@ protected override void Write(LogEventInfo logEvent)
...
 
             var exception = logEvent.Exception;
-            var formatted = Layout.Render(logEvent);
+            var formatted = RenderLogEvent(Layout, logEvent);
             var template = logEvent.Message;
 
-            var contextProps = GetAllProperties(logEvent);
-
             var shouldOnlyLogExceptions = exception == null && IgnoreEventsWithNoException;
+            var shouldIncludeProperties = ContextProperties?.Count > 0 || ShouldIncludeProperties(logEvent);
```
Comment (bruno-garcia) [inline src/Sentry.NLog/SentryTarget.cs]: is `IncludeEventProperties` irrelevant now? If so, please remove it.
Do we need to override `ShouldIncludeProperties` here?
We're preparing for 2.0.0 so breaking changes are welcome.

File: test/Sentry.NLog.Tests/SentryTargetTests.cs [hunk 4]
```diff
@@ -103,7 +103,9 @@ public void Can_configure_from_xml_file()
                     </targets>
                 </nlog>";
 
-            var c = XmlLoggingConfiguration.CreateFromXmlString(configXml);
+            var stringReader = new System.IO.StringReader(configXml);
```
Comment (bruno-garcia) [inline test/Sentry.NLog.Tests/SentryTargetTests.cs]: What's the purpose of this change?

File: src/Sentry.NLog/LogLevelExtensions.cs (hunk 2 above)
Comment (snakefoot) [reply src/Sentry.NLog/LogLevelExtensions.cs]: Well `null` is not a good default value. Prefer something more sensible. Though it cannot be tested :)

File: src/Sentry.NLog/SentryTarget.cs (hunk 3 above)
Comment (snakefoot) [reply src/Sentry.NLog/SentryTarget.cs]: `IncludeEventProperties` lives on `TargetWithContext`. Cannot be removed.  `TargetWithContext.ShouldIncludeProperties()` checks `IncludeEventProperties`.

File: test/Sentry.NLog.Tests/SentryTargetTests.cs (hunk 4 above)
Comment (snakefoot) [reply test/Sentry.NLog.Tests/SentryTargetTests.cs]: I'm downgrading the NLog-version from NLog 4.6 to 4.5.11. This helper-method is not available in 4.5.11

**Comment by snakefoot** [comment]
> You change introduced a couple of branches uncovered by tests. Could you please test those out before we can merge it?

Pretty sure those code-branches was not tested before. But I can try to make some more unit-tests.

File: src/Sentry.NLog/LogLevelExtensions.cs (hunk 2 above)
Comment (snakefoot) [reply src/Sentry.NLog/LogLevelExtensions.cs]: Can revert the change if you want the `null`-value.

File: src/Sentry.Log4Net/Sentry.Log4Net.csproj (hunk 1 above)
Comment (snakefoot) [reply src/Sentry.Log4Net/Sentry.Log4Net.csproj]: Can revert the change if you want?

File: src/Sentry.NLog/LogLevelExtensions.cs (hunk 2 above)
Comment (bruno-garcia) [reply src/Sentry.NLog/LogLevelExtensions.cs]: Unless we know what's the side effect here I'd rather leave it as it was.
I agree generally but in this case, a value that cannot be mapped might just not fit `Info`. It would be a value created in the future which we don't support (i.e: `None`).

File: src/Sentry.Log4Net/Sentry.Log4Net.csproj (hunk 1 above)
Comment (bruno-garcia) [reply src/Sentry.Log4Net/Sentry.Log4Net.csproj]: Please do. If you want to open another PR just for that I'd be happy to merge. Otherwise I'll open one because this was an oversight.

File: src/Sentry.NLog/SentryTarget.cs (hunk 3 above)
Comment (bruno-garcia) [reply src/Sentry.NLog/SentryTarget.cs]: Got it. Thanks for clarifying

File: test/Sentry.NLog.Tests/SentryTargetTests.cs (hunk 4 above)
Comment (bruno-garcia) [reply test/Sentry.NLog.Tests/SentryTargetTests.cs]: LGTM

File: src/Sentry.Log4Net/Sentry.Log4Net.csproj (hunk 1 above)
Comment (snakefoot) [reply src/Sentry.Log4Net/Sentry.Log4Net.csproj]: Reverted the changes. And forced pushed them away.

File: src/Sentry.NLog/LogLevelExtensions.cs (hunk 2 above)
Comment (snakefoot) [reply src/Sentry.NLog/LogLevelExtensions.cs]: Reverted the changes. And forced pushed them away.

**Comment by snakefoot** [comment]
> Could you please test those out before we can merge it?

Test coverage has now been improved.

**Comment by bruno-garcia** [comment]
Thanks @snakefoot
